lib/sidekiq/transaction_guard.rb

```diff
@@ -46,21 +48,23 @@ def notify(&block)
 
       # Return the block set as the notify handler with a call to `notify`.
       def notify_block
-        @notify ||= nil
+        @notify
       end
 
       # Add a class that maintains it's own connection pool to the connections
       # being monitored for open transactions. You don't need to add `ActiveRecord::Base`
       # or subclasses. Only the base class that establishes a new connection pool
       # with a call to `establish_connection` needs to be added.
       def add_connection_class(connection_class)
-        @connection_classes ||= Set.new
-        @connection_classes << connection_class
+        @lock.synchronize{ @connection_classes << connection_class }
       end
 
       # Return true if any connection is currently inside of a transaction.
       def in_transaction?
-        connection_classes = @lock.synchronize{ @connection_classes.dup }
+        connection_classes = [ActiveRecord::Base]
```

Comment:
Wouldn't it be simpler to do some version of adding `ActiveRecord::Base` to `@connection_classes` rather than this approach of creating a new array every time?

Reply:
To be thread safe, I need to duplicate the connection classes set inside a synchronized block. However, the default case will usually be it is just an array with ActiveRecord::Base in it. This way, I don't need to synchronize. As an added bonus, we don't need to explicitly require `active_record` because we're no longer referencing ActiveRecord classed during initialization.

Reply:
Okay, concat just _feels_ wrong, so usually I'd think of something like this, but honestly anything is fine:
```ruby
connection_classes = []
unless @connection_classes.empty?
  connection_classes = @lock.synchronize{ @connection_classes.to_a }
end
connection_classes << ActiveRecord::Base
```

Reply:
Actually, what would be a case where the class has a `connection_pool` but doesn't extend `ActiveRecord::Base`? Also, are there any particular considerations we should be making for `ApplicationRecord` in Rails 5?